### PR TITLE
Add tooltips to show keyboard shortcuts 

### LINKED ISSUE #665: Additionally show the keyboard shortcuts for buttons in their tooltip
Most applications show a description of the action of a button, along with the assigned keyboard shortcut (if any)  in a tooltip, when hovering. As a first-time user, I was expecting this, especially since the F11-Fullscreen shortcut "did not work". (See separate issue https://github.com/kiwix/kiwix-desktop/issues/304)

Here's how it looks like currently:
hovering-tooltip
_The tooltip on hover is displayed without keyboard shortcut information_

Hovever, in the menu, however, the shortcut (CRTL+T in this case), is displayed properly.

Expected behaviour: The tooltip should display the keyboard shortcut along the text.
hovering-tooltip-with-shortcut
_The proposed tooltip with keyboard shortcut information_

Note: I experience this on a fresh, updated Ubuntu 20.04, using Version 2.0.5

### PR DESCRIPTION
**Pull request by juuz0**
Fixes #665 

Tool tips for the following have been added -
New Tab (Ctrl + T)
Close Tab (Ctrl + F4)
Reading List (Ctrl + Shift + 2)
Open File (Ctrl + O)
Toggle fullscreen (F11)

image

### CODE REVIEW

**Comment by kelson42** [comment]
@juuz0 Please don't merge master in your branch. Instead make a rebase.

**Comment by juuz0** [comment]
Would this work?

**Comment by kelson42** [comment]
@juuz0 Yes, we are a bit picky about how the feature branches revision log should look like… all this to keep a clean revision log once the PR will be merged.

File: src/kiwixapp.cpp
Comment (mgautierfr) [inline src/kiwixapp.cpp:319]: The sequence for AddTab may not be `Ctrl+T` depending of the platform.
We must not hardcode this.
The best way is to get the string from https://doc.qt.io/qt-5/qkeysequence.html#toString
This way we will always display the right shortcut to the user.

I also wonder if the property `shortcutVisibleInContextMenu` of the action itself (https://doc.qt.io/qt-5/qaction.html#shortcutVisibleInContextMenu-prop) could help us also.

**Review by mgautierfr — CHANGES_REQUESTED** [review]
As said in the code comment, we must not hardcode the shortcut displayed to the user.

**Comment by juuz0** [comment]
On it, should I push the changes in this branch (issue665) only?

**Comment by kelson42** [comment]
@juuz0 yes please, where else?

**Comment by kelson42** [comment]
@juuz0 thank you for the fixes, is this ready now for a new review?

**Comment by juuz0** [comment]
@kelson42 Yes!

**Review by mgautierfr — CHANGES_REQUESTED** [review]
The code is functionally good but difficult to read and redundant with other configuration of the action (we are writing twice the sequence and the label of the action).

It would be better to addapt the existing macros and have a new macro `CREATE_ACTION_SHORTCUT`/`CREATE_ACTION_ICON_SHORTCUT` which create the action, set the shortcut and the tooltip for us in "one" shoot.

**Comment by kelson42** [comment]
@juuz0 Would you be able to follow @mgautierfr advices and make the necessary improvements?

**Comment by juuz0** [comment]
@kelson42 @mgautierfr  I think this is what was asked :)

File: src/kiwixapp.cpp
Comment (mgautierfr) [inline src/kiwixapp.cpp:380]: Could we keep a `CREATE_ACTION` for action without shortcut ?

File: src/kiwixapp.cpp
Comment (mgautierfr) [inline src/kiwixapp.cpp:288]: We probably don't want to set all icons invisible in menu.
We were setting the icon invisible only for closeTabAction.
Now we are setting this for all actions.

**Review by mgautierfr — CHANGES_REQUESTED** [review]
We are almost good. This is far better.
There is still two small changes.

For the macros, we can keep the following ones :
```cpp
#define CREATE_ACTION(ID, TEXT) \
    mpa_actions[ID] = new QAction(TEXT)
#define SET_SHORTCUT(ID, TEXT, SHORTCUT) \
    mpa_actions[ID]->setShortcut(SHORTCUT); \
    mpa_actions[ID]->setToolTip(TEXT + " (" + KeySequence(SHORTCUT).toString() + ")" )
#define CREATE_ACTION_SHORTCUT(ID, TEXT, SHORTCUT) \
    CREATE_ACTION(ID, TEXT); \
	SET_SHORTCUT(ID, TEXT, SHORTCUT)
#define CREATE_ACTION_ICON_SHORTCUT(ID, ICON, TEXT, SHORTCUT) \
    mpa_actions[ID] = new QAction(QIcon(":/icons/" ICON ".svg"), TEXT); \
    SET_SHORTCUT(ID, TEXT, SHORTCUT)
```

File: src/kiwixapp.cpp
Comment (juuz0) [reply src/kiwixapp.cpp:288]: Yeah, I kind of assumed all should be invisible in menu, will fix this thanks!

File: src/kiwixapp.cpp
Comment (mgautierfr) [inline src/kiwixapp.cpp:291]: Tab instead of spaces.

**Review by mgautierfr — CHANGES_REQUESTED** [review]
On very small tiny change and we are good.

You can squash all commits into one for this PR. The change is pretty self contained.

**Comment by juuz0** [comment]
I am not sure on how that tab problem happened but hopefully this is fine now.

**Comment by kelson42** [comment]
@juuz0 Thx for your work, merging now.

**Pull request merged**
